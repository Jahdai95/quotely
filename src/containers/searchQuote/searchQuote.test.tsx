import { render, screen, fireEvent } from "@testing-library/react";
import { describe, test, expect, it } from "vitest";
import userEvent from "@testing-library/user-event";

import SearchQuote from "./searchQuote";
import { QuotesProvider } from "../../providers/quotes.provider";

describe("Should render AddQuote component", () => {
  const getRender = () => {
    return render(
      <QuotesProvider>
        <SearchQuote />
      </QuotesProvider>
    );
  };
  test("Should to search quote", async () => {
    getRender();

    const input = screen.getByTestId("input-search-quote");
    await userEvent.click(input);
    fireEvent.change(input, {
      target: { value: "palabra" },
    });
    expect(input).toHaveValue("palabra");

    const button = screen.getByTestId("button-clear-input");
    await userEvent.click(button);
  });

  it("Should macth snapshot", () => {
    const { container } = getRender();
    expect(container).toMatchSnapshot();
  });
});
