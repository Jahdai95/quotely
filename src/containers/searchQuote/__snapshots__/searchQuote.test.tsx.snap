// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render AddQuote component > Should macth snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-k008qs"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-r68sd5-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="input-search-quote"
          id=":r3:"
          placeholder="Escribe para buscar una cita"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-w4cd9x"
          >
            <span
              aria-hidden="true"
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      aria-label="Limpiar"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1svfwy9-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      data-testid="button-clear-input"
      id=":r5:"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="NotInterestedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.42 0-8-3.58-8-8 0-1.85.63-3.55 1.69-4.9L16.9 18.31C15.55 19.37 13.85 20 12 20m6.31-3.1L7.1 5.69C8.45 4.63 10.15 4 12 4c4.42 0 8 3.58 8 8 0 1.85-.63 3.55-1.69 4.9"
        />
      </svg>
    </button>
  </div>
  <div
    class="MuiBox-root css-1mx2345"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-k1p3sh-MuiPaper-root-MuiAlert-root"
      role="alert"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiAlert-icon css-vab54s-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="PriorityHighIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="19"
            r="2"
          />
          <path
            d="M10 3h4v12h-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-zioonp-MuiAlert-message"
      >
        <p
          class="quicksand-200"
        >
          No hay citas para mostrar agregÃ¡ alguna ðŸ˜œ
        </p>
      </div>
    </div>
  </div>
</div>
`;
